src/vs/workbench/parts/search/browser/searchResultsView.ts

```diff
@@ -354,3 +359,63 @@ export class SearchFilter implements IFilter {
 		return !(element instanceof FileMatch || element instanceof FolderMatch) || element.matches().length > 0;
 	}
 }
+
+export class SearchTreeController extends WorkbenchTreeController {
```

Comment:
This works fine and we have it for instance in debug land.
However this is an old way to do this - it is a sort of bridge between comands and actions.

Since all your things could be commands this can than be much simpliar without any action items and things like that.
I suggest to look here for an inspiration on how to do it if all you guys are commands
https://github.com/Microsoft/vscode/blob/roblou/searchContextMenu/src/vs/workbench/parts/files/electron-browser/views/openEditorsView.ts#L312

Reply:
I'm guessing you're referring to what I do inside showContextMenu, and not the Controller itself? I see that I can use `fillInActions` which is simpler. But still seems action-oriented...

Reply:
@roblourens yeah, you have to use the controller since you are a tree. However once you tackle the debt item to convert aciton to commands this might simpllify itselft.